src/base/misc/fatfs.c

```diff
@@ -1585,19 +1588,13 @@ void build_boot_blk(fatfs_t *f, unsigned char *b)
   b[0x44] = DOS_HELPER_INT;
 
   /*
-   * IO.SYS from MS-DOS 7 normally re-uses the boot block's error message. We
-   * could give it a distinct one here (simply have t_o point to it).
-   *
-   * But don't forget, it's not a simple ASCIIZ string!!!
-   *
-   * NOTE: There is a discrepancy between IO.SYS's and the bootblock's
-   * interpretation of this address. IO.SYS expects t_o to be
-   * relative to 0, but the value actually stored is relative to 0x7c00.
-   * (Leading IO.SYS to display no error message.)
-   * I will assume IO.SYS to be correct for now.
+   * IO.SYS from MS-DOS 7 normally re-uses the boot block's error message.
+   * Please note that the address points to four bytes before the string to
+   * be displayed so we pad beforehand. Presumably this data would be
+   * meaningful, but as yet we do not know what it might be used for.
    */
   t_o = 0x48;
-  snprintf((char *)b + t_o, eos - t_o, msg_f, f->dir);
+  snprintf((char *)b + t_o, eos - t_o, "\x04\x03\x02\x01" MSG_F, f->dir);
```

Comment:
> Do you have a succinct method of doing the same without the issue with embedded nul?

No, but the problem is not only nul.
Percent sign will lead to a crash, for example.
But OK, I guess I can leave it alone.

Could you review the
https://github.com/stsp/dosemu2/commit/cee3f8d8003dca43f214d747b5b958d43862e138
please, btw?